internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -1766,31 +1831,47 @@ func (r *ClusterDeploymentsReconciler) updateStatus(ctx context.Context, log log
 			clusterInstall.Status.UserManagedNetworking = c.UserManagedNetworking
 			clusterInstall.Status.PlatformType = getPlatformType(c.Platform)
 			status := *c.Status
-			var err error
-			err = r.populateEventsURL(log, clusterInstall, c)
+
+			err := r.populateEventsURL(log, clusterInstall, c)
 			if err != nil {
 				return ctrl.Result{Requeue: true}, nil
 			}
 			err = r.populateLogsURL(log, clusterInstall, c)
 			if err != nil {
 				return ctrl.Result{Requeue: true}, nil
 			}
-			var registeredHosts, approvedHosts, unsyncedHosts int
+			var approvedHosts, unsyncedHosts int
 			if status == models.ClusterStatusReady {
-				registeredHosts, approvedHosts, err = r.getNumOfClusterAgents(c)
+				_, approvedHosts, err = r.getNumOfClusterAgents(c)
 				if err != nil {
 					log.WithError(err).Error("failed to fetch cluster's agents")
 					return ctrl.Result{Requeue: true}, nil
 				}
-				agents, err := findAgentsByAgentClusterInstall(r.Client, ctx, log, clusterInstall)
+				var agents []*aiv1beta1.Agent
+				agents, err = findAgentsByAgentClusterInstall(r.Client, ctx, log, clusterInstall)
 				if err != nil {
 					log.WithError(err).Error("failed to fetch ACI's agents")
 					return ctrl.Result{Requeue: true}, nil
 				}
+
+				mastersCountPtr, workersCountPtr, err = getHostSuggestedRoleCount(r.ClusterApi, *c.ID)
+				if err != nil {
+					log.WithError(err).Error("failed to fetch host suggested role count")
+					// proceed to update the conditions (the counts will be 0)
+				}
+
 				unsyncedHosts = getNumOfUnsyncedAgents(agents)
 				log.Debugf("Updating ACI conditions, found %d unsynced agents out of total of %d agents", unsyncedHosts, len(agents))
 			}
-			clusterRequirementsMet(clusterInstall, status, registeredHosts, approvedHosts, unsyncedHosts)
+
+			clusterRequirementsMet(
+				clusterInstall,
+				status,
+				approvedHosts,
+				unsyncedHosts,
+				int(swag.Int64Value(mastersCountPtr)),
+				int(swag.Int64Value(workersCountPtr)),
```

Comment:
Will these cause any issues if `masterCountPtr` or `workersCountPtr` is nil?

Reply:
If these are nil then they will be 0, will probably show something like - `"The cluster currently requires exactly 3 master agents and 2 worker agents, but currently registered 0 master agents and 0 worker agents"`. I son't see any other option if we want to update status in this case, as we don't want to show custom message in this case right ?

Reply:
Cool just confirming it doesn't cause issues. That sounds good, thanks!